File: coding/04th January 2025/Find Pivot Index.java
//Find Pivot Index
class Solution 
{
    public int pivotIndex(int[] nums) 
    {
        int totalsum = 0,lsum = 0;
        for(int i = 0; i < nums.length; i++)
            totalsum = totalsum + nums[i];
        for(int i = 0; i < nums.length; i++)
        {
            totalsum = totalsum - nums[i];
            if(totalsum == lsum)
                return i;
            lsum = lsum + nums[i];
        }
        return -1;
    }
}

File: Coding in Python/21st January 2026/Find Pivot Index.py
#Find Pivot Index
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        lsum = 0
        tsum = 0
        for i in range(0, len(nums)):
            tsum += nums[i]
        for i in range(0, len(nums)):
            tsum -= nums[i]
            if(tsum == lsum):
                return i
            lsum += nums[i]
        return -1
